### PR TITLE
Avoid cloning function in PatchEntryPointMutate

### PR DESCRIPTION
**Pull request by trenouf**
PatchEntryPointMutate was adding entry args to a shader by cloning the
function into a new function with the args. This commit changes it to
move the code into the new function instead, which is cheaper.

Change-Id: I4b6cc9fa19edf0f569968da3e7059f25343f0b8b

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 24c2f65aff4b8171f35f758a524579c931d34839
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev ca6804680a807d27b7e9628d78165a0a8607fc0f</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev b797a4faec565ad9a3ae987eb42fcf7c943203b1</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 53692d985a53a336e07907c2d4b86bf2deb66908</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 39862/39863 (100.0%)</li>
<li><details><summary>Failed: 1/39863 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.spirv_assembly.instruction.graphics.variable_pointers.multi_buffer_read_only_graphics.opselect_arrays_first_input_geom
Stack trace: Timeout or crahsed
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchEntryPointMutate.cpp [hunk 1]
```diff
@@ -110,14 +110,22 @@ void PatchEntryPointMutate::processShader() {
...
+  // Create new function, empty for now.
   Function *entryPoint = Function::Create(entryPointTy, GlobalValue::ExternalLinkage, "", m_module);
   entryPoint->setCallingConv(origEntryPoint->getCallingConv());
-  entryPoint->addFnAttr(Attribute::NoUnwind);
   entryPoint->takeName(origEntryPoint);
 
-  ValueToValueMapTy valueMap;
-  SmallVector<ReturnInst *, 8> retInsts;
-  CloneFunctionInto(entryPoint, origEntryPoint, valueMap, false, retInsts);
+  // Transfer code from old function to new function.
```
Comment (amdrexu) [inline lgc/patch/PatchEntryPointMutate.cpp:118]: This is a good change. I have one question: how to copy blocks into multiple functions. Does it mean we couldn't simply just set new parents for such blocks?

Comment (trenouf) [reply lgc/patch/PatchEntryPointMutate.cpp:118]: If you want to copy the same block into N different places, then you will need at least N-1 clones. Is that what you mean?

Comment (amdrexu) [reply lgc/patch/PatchEntryPointMutate.cpp:118]: Yes, this is what I mean. If we just move the blocks from one parent to another, it is cheap. If you want N copies of those blocks for N parents, cloning is still needed. I just want to confirm this.

**Pull request merged**